modules/ossm-threescale-webassembly-module-examples-for-credentials-use-cases.adoc

```diff
@@ -0,0 +1,236 @@
+// Module included in the following assembly:
+//
+// service_mesh/v2x/ossm-threescale-webassembly-module.adoc
+
+[id="ossm-threescale-webassembly-module-examples-for-credentials-use-cases_{context}"]
+= The 3scale WebAssembly module examples for `credentials` use cases
+
+You will spend most of your time applying configuration steps to obtain credentials in the requests to your services.
+
+The following are `credentials` example, which you can modify to adapt to specific use cases.
+
+You can combine them all, although when you specify multiple source objects with their own `lookup queries`, they are evaluated in order until one of them successfully resolves.
+
+[id="api-key-in-query-string-parameters_{context}"]
+== API key (`user_key`) in query string parameters
+This looks up a `user_key` in a query string parameter or header of the same name:
+
+[source,yaml]
+----
+credentials:
+  user_key:
+    - query_string:
+        keys:
+          - user_key
+    - header:
+        keys:
+          - user_key
+----
+
+[id="application-id-and-key_{context}"]
+== Application ID and `key`
+The following example looks up a `app_key` and `app_id` in a query or headers.
+
+[source,yaml]
+----
+credentials:
+  app_id:
+    - header:
+        keys:
+          - app_id
+    - query_string:
+        keys:
+          - app_id
+  app_key:
+    - header:
+        keys:
+          - app_key
+    - query_string:
+        keys:
+          - app_key
+----
+
+[id="authorization-header_{context}"]
+== Authorization header
+A request includes an `app_id` and `key` in an `authorization` header. If there is at least one or two values outputted at the end, then you can assign the `app_key`.
+
+The resolution here will assign the `application key` if there is one or two output at the end.
```

Comment:
Is `application key` the same as `app_key` or `key` mentioned previously?

Also I see a line up used the phrasing "outputted at the end" and here it's "output at the end". I think either is valid, but maybe choose one or the other.

Reply:
I think `key` is authorization key, but yes application key is app_key.

@rahulanand16nov might confirm the former.

Reply:
Yup, `application key` is `app_key`.